.github/workflows/check.yaml

```diff
@@ -136,6 +136,15 @@ jobs:
           TEST_JUJU3: "1"  # https://github.com/openstack-charmers/zaza/pull/653
           TEST_JUJU_CHANNEL: ${{ matrix.juju-channel }}
 
+      - name: Determine system architecture
+        run: echo "SYSTEM_ARCH=$(uname -m)" >> $GITHUB_ENV
+
+      - name: Upload the tested charms
+        uses: actions/upload-artifact@v4
+        with:
+          name: built_charms_${{ env.SYSTEM_ARCH }}
+          path: ./*.charm
```

Comment:
Will this work even if there are multiple charm files?

Reply:
Yes, check this test workflow: https://github.com/canonical/charm-nrpe/actions/runs/13559772104/job/37900860316

Reply:
in the download action, it should contains: `CHARM_NAMES=nrpe_ubuntu@20.04-amd64.charm,nrpe_ubuntu@20.04-arm64.charm,nrpe_ubuntu@22.04-amd64.charm,nrpe_ubuntu@22.04-arm64.charm`